runtime/src/accounts_db.rs

```diff
@@ -2619,6 +2623,16 @@ impl AccountsDb {
         }
         remove_storages_elapsed.stop();
 
+        for s in &all_removed_slot_storages {
+            for (_store_id, store) in s.read().unwrap().iter() {
+                if !store.unref_done.swap(true, Ordering::Relaxed) {
```

Comment:
@sakridge here's my first attempt. All tests pass with the exception of a race condition I'm debugging in:
test_store_scan_consistency_unrooted

Reply:
I see. So if we may go through the unref path twice in some cases. This would prevent that by keeping a bool per store to prevent it.

Reply:
All runtime tests now pass locally. I wasn't handling `AccountStorageEntry::recycle`.